test/support/console_test_case.rb

```diff
@@ -50,133 +47,38 @@ def create_message fail_msg, test_info
       MSG
     end
 
-    TestInfo = Struct.new(:queue, :mode, :prompt_pattern, :remote_info,
-                          :backlog, :last_backlog, :internal_info, :failed_process)
-
-    RemoteInfo = Struct.new(:r, :w, :pid, :sock_path, :port, :reader_thread, :debuggee_backlog)
-
-    MULTITHREADED_TEST = !(%w[1 true].include? ENV['RUBY_DEBUG_TEST_DISABLE_THREADS'])
-
     def debug_code(program, remote: true, &test_steps)
-      prepare_test_environment(program, test_steps) do
-        if remote && !NO_REMOTE && MULTITHREADED_TEST
-          begin
-            th = [
-              new_thread { debug_code_on_local },
-              new_thread { debug_code_on_unix_domain_socket },
-              new_thread { debug_code_on_tcpip },
-            ]
-
-            th.each do |t|
-              if fail_msg = t.join.value
-                th.each(&:kill)
-                flunk fail_msg
+      Timeout.timeout(60) do
```

Comment:
Should we respect `TIMEOUT_SEC` for this timeout?

Reply:
Yeah good idea. There shouldn't be any test taking more than 10 seconds to run atm.